recipes/openssl/all/openssl_extra.cmake

```diff
@@ -0,0 +1,45 @@
+# This script should only run when included in a FindXXX.cmake module
+if(NOT OpenSSL_LIB_DIRS)
+    return()
+endif()
+
+# library names of crypto and ssl libraries
+set(CONAN_CRYPTO_LIBRARY "@CONAN_CRYPTO_LIBRARY@")
+set(CONAN_SSL_LIBRARY "@CONAN_SSL_LIBRARY@")
+
+# absolute paths of crypto and ssl libraries
+find_library(CONAN_CRYPTO_ABSOLUTE_LIBRARY NAME ${CONAN_CRYPTO_LIBRARY} PATHS ${OpenSSL_LIB_DIRS} NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
+find_library(CONAN_SSL_ABSOLUTE_LIBRARY NAME ${CONAN_SSL_LIBRARY} PATHS ${OpenSSL_LIB_DIRS} NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
+
+# https://cmake.org/cmake/help/latest/module/FindOpenSSL.html
```

Comment:
It is probably a good idea to call find_package_handle_standard_args (https://cmake.org/cmake/help/latest/module/FindPackageHandleStandardArgs.html) to report the error if the files could not be found and set some of the variables that are set in the next few lines.

There is more documentation on regarding writing finders in https://gitlab.kitware.com/cmake/community/wikis/doc/tutorials/How-To-Find-Libraries.

Reply:
Good idea about `find_package_handle_standard_args`.